lib/kafka/consumer.rb

```diff
@@ -340,12 +340,12 @@ def fetch_batches(min_bytes:, max_wait_time:)
             @logger.warn "Partition #{topic}/#{partition} is currently paused, skipping"
           else
             @logger.debug "Fetching batch from #{topic}/#{partition} starting at offset #{offset}"
-            operation.fetch_from_partition(topic, partition, offset: offset, max_bytes: max_bytes)
+            @operation.fetch_from_partition(topic, partition, offset: offset, max_bytes: max_bytes)
           end
         end
       end
 
-      operation.execute
+      @operation.execute
```

Comment:
I think memoizing the operation opens us up to a range of new problems. For instance, the arguments to `fetch_batches` would be ignored on all but the first call...

Is there an alternative?